packages/@aws-cdk/aws-rds/lib/instance.ts

```diff
@@ -305,6 +305,13 @@ export interface DatabaseInstanceNewProps {
    */
   readonly vpcPlacement?: ec2.SubnetSelection;
 
+  /**
+   * The security group to assign to the DB instance.
+   *
+   * @default - a new security group is created
+   */
+  readonly securityGroup?: ec2.ISecurityGroup;
```

Comment:
You can have an array of SecurityGroups for instance, right? So this should be `readonly securityGroups: ec2.ISecurityGroup[]`, correct?

Reply:
Agree but this is to be consistent with other packages in the CDK, examples are ECS Service, RDS Cluster, ELBV2 Load balancers, etc.

Reply:
The only exception is the AWS CodeBuild project.

Reply:
I discussed this with the team, and we came to the conclusion that array is the way to go, and the other places that don't allow arrays are actually the ones that are wrong (and they will have to change, like ECS does in #3985 ).

So, please change it to `readonly securityGroups: ISecurityGroup[]`.

Thanks!

Reply:
Done. Note there's now a `BREAKING CHANGE` documented in the PR description, someone of the team will need to squash this PR because `mergify` won't take it into account.

Reply:
Is the breaking change removing the `securityGroupId` property?

Reply:
BREAKING CHANGE: `securityGroup: ec2.ISecurityGroup` is now `securityGroups: ec2.ISecurityGroup[]` in `DatabaseInstanceAttributes`, removed `securityGroupId` from `IDatabaseInstance`

Reply:
I think I'm missing something then. Wasn't it possible to use existing security groups with Instance before this PR? When you could pass in a single security group?

Reply:
Oh, I think I see it now. It's about allowing to use an existing security group for a *new* Instance, right? Before, it was only possible to pass it for imported Instances?

Reply:
Yes, exactly